common/stats/dir_monitor.go

```diff
@@ -0,0 +1,104 @@
+package stats
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var NopDirMonitor *DirMonitor = NewDirMonitor([]MonitorDir{})
+
+// MonitorDir a directory to monitor and a shortname (suffix) for reporting the stat
+type MonitorDir struct {
+	Directory  string // the directory to monitor
+	StatSuffix string // the suffix to use on the commandDirUsage_kb stat
+}
+
+// DirMonitor monitor disk usage for selected directories
+type DirMonitor struct {
+	dirs       []MonitorDir
+	startSizes []int64
+	endSizes   []int64
+}
+
+// NewDirMonitor return a DirMonitor
+func NewDirMonitor(dirs []MonitorDir) *DirMonitor {
+	dm := &DirMonitor{dirs: dirs, startSizes: make([]int64, len(dirs)), endSizes: make([]int64, len(dirs))}
+	for i := range dirs {
```

Comment:
Looks like we did this half way, I meant even including the startSize and endSize in DirMonitor, so you don't need this for loop to initialize these values, and you don't have to index different slices anywhere in this file.